Examples/Tests/MacroExamples/Implementation/Peer/AddAsyncMacroTests.swift

```diff
@@ -0,0 +1,131 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import MacroExamplesImplementation
+import SwiftSyntaxMacros
+import SwiftSyntaxMacrosTestSupport
+import XCTest
+
+final class AddAsyncMacroTests: XCTestCase {
+  private let macros = ["AddAsync": AddAsyncMacro.self]
+
+  func testExpansionTransformsFunctionWithResultCompletionToAsyncThrows() {
+    assertMacroExpansion(
+      #"""
+      @AddAsync
+      func c(a: Int, for b: String, _ value: Double, completionBlock: @escaping (Result<String, Error>) -> Void) -> Void {
+        completionBlock(.success("a: \(a), b: \(b), value: \(value)"))
+      }
+      """#,
+      expandedSource: #"""
+        func c(a: Int, for b: String, _ value: Double, completionBlock: @escaping (Result<String, Error>) -> Void) -> Void {
+          completionBlock(.success("a: \(a), b: \(b), value: \(value)"))
+        }
+
+        func c(a: Int, for b: String, _ value: Double) async throws -> String {
+          try await withCheckedThrowingContinuation { continuation in
+            c(a: a, for: b, value) { returnValue in
+
+              switch returnValue {
+          case .success(let value):
+              continuation.resume(returning: value)
+          case .failure(let error):
+              continuation.resume(throwing: error)
+          }
+
+            }
+          }
```

Comment:
You should be able to fix this formatting by changing the implementation of `AddAsyncMacro` to 

```swift
    let switchBody: ExprSyntax =
      """
            switch returnValue {
            case .success(let value):
              continuation.resume(returning: value)
            case .failure(let error):
              continuation.resume(throwing: error)
            }
      """

    let newBody: ExprSyntax =
      """

        \(raw: isResultReturn ? "try await withCheckedThrowingContinuation { continuation in" : "await withCheckedContinuation { continuation in")
          \(raw: funcDecl.name)(\(raw: callArguments.joined(separator: ", "))) { \(raw: returnType != nil ? "returnValue in" : "")

      \(raw: isResultReturn ? switchBody : "continuation.resume(returning: \(raw: returnType != nil ? "returnValue" : "()"))")

          }
        }

      """
```

Reply:
Done in https://github.com/apple/swift-syntax/pull/2163/commits/39032ab805de40a7a4c7416a71a3f7704662a8ce. Thanks!